File: 0917-reverse-only-letters/0917-reverse-only-letters.java
class Solution {
    public String reverseOnlyLetters(String s) {
        //Convert string to array
        char arr[] = s.toCharArray();
        //Using two pointer approach
        int left = 0;
        int right = arr.length-1;
        while(left<right){
            //Skip non-letters from left
            while(left<right &&!Character.isLetter(arr[left])){
                left++;
            }
            //skip non-letters from right
            while(left<right &&!Character.isLetter(arr[right])){
                right--;
            }
            //Swap letters
            char temp =arr[left];
            arr[left] = arr[right];
            arr[right] = temp;
            left++;
            right--;
        }
        return new String(arr);
    }
}

File: 0917-reverse-only-letters/0917-reverse-only-letters.py
class Solution(object):
    def reverseOnlyLetters(self, s):
        # convert string to list (since strings are immutable in Python)
        arr = list(s)

        left = 0
        right = len(arr) - 1

        while left < right:
            # skip non-letters from left
            while left < right and not arr[left].isalpha():
                left += 1

            # skip non-letters from right
            while left < right and not arr[right].isalpha():
                right -= 1

            # swap letters
            arr[left], arr[right] = arr[right], arr[left]

            left += 1
            right -= 1

        return "".join(arr)